src/plugins/liveobjects/syncliveobjectsdatapool.ts

```diff
@@ -1,29 +1,24 @@
 import type BaseClient from 'common/lib/client/baseclient';
 import type RealtimeChannel from 'common/lib/client/realtimechannel';
-import { LiveCounterData } from './livecounter';
-import { LiveMap } from './livemap';
-import { LiveObjectData } from './liveobject';
 import { LiveObjects } from './liveobjects';
-import { MapSemantics, StateMessage, StateObject } from './statemessage';
+import { StateMessage, StateObject } from './statemessage';
 
 export interface LiveObjectDataEntry {
-  objectData: LiveObjectData;
-  regionalTimeserial: string;
+  stateObject: StateObject;
   objectType: 'LiveMap' | 'LiveCounter';
 }
 
 export interface LiveCounterDataEntry extends LiveObjectDataEntry {
-  created: boolean;
   objectType: 'LiveCounter';
 }
 
 export interface LiveMapDataEntry extends LiveObjectDataEntry {
   objectType: 'LiveMap';
-  semantics: MapSemantics;
 }
 
 export type AnyDataEntry = LiveCounterDataEntry | LiveMapDataEntry;
 
+// TODO: investigate if this class is still needed after changes with createOp. objects are now initialized from the stateObject and this class does minimal processing
```

Comment:
Could the copy of the pool for sync just be a `LiveObjectsPool` instance? Maybe not, in order to preserve data refs?

Reply:
it's not even a copy of anything at this point, it just buffers incoming state object ops during sync and that's all.
the caller class, `LiveObjects`, then just gets the full array of buffers state objects and "applies the sync" by actully creating new objects/overriding existing ones.

I'll keep this `SyncLiveObjectsDataPool` class  around for a bit, until all api is in and we're a bit closer to the stable protocol. If that point it still does the same thing as right now, I'll just move buffering to the caller `LiveObjects` class (it already does buffering of state ops, so seems reasonable to also buffer state sync ops).